From 2bde763d5d4df3ab12ad7c4a1a231c363ed6e54c Mon Sep 17 00:00:00 2001
From: vipinc <vipinc@ami.com>
Date: Wed, 18 Oct 2023 11:47:50 +0530
Subject: [PATCH] PEF: Update install location for systemd service .

This patch update the install location to resovle the build error
and update discreate sensor interface.

Signed-off-by: vipinc <vipinc@ami.com>
---
 CMakeLists.txt       | 4 +++-
 include/sdrutils.hpp | 3 ++-
 2 files changed, 5 insertions(+), 2 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index ae4d52e..61fc444 100755
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -39,7 +39,9 @@ target_link_libraries(pef-event-filtering boost_coroutine)
 target_link_libraries(pef-event-filtering sdbusplus -lstdc++fs)
 target_link_libraries(pef-event-filtering ${Boost_LIBRARIES})
 target_link_libraries(pef-event-filtering ${SNMP_LIBRARY})
+pkg_get_variable(SYSTEMD_SYSTEM_UNIT_DIR systemd systemdsystemunitdir)
 
+ 
 
 link_directories(${EXTERNAL_INSTALL_LOCATION}/lib)
 
@@ -50,7 +52,7 @@ set(CONFIG_FILES
     ${PROJECT_SOURCE_DIR}/pef_configurations/pef-alert-manager.json)
 
 install(TARGETS pef-configuration DESTINATION bin)
-install(FILES ${SERVICE_FILES} DESTINATION /lib/systemd/system/)
+install(FILES ${SERVICE_FILES} DESTINATION ${SYSTEMD_SYSTEM_UNIT_DIR})
 install(FILES ${CONFIG_FILES} DESTINATION /usr/share/pef-alert-manager/)
 
 install(TARGETS pef-event-filtering DESTINATION bin)
diff --git a/include/sdrutils.hpp b/include/sdrutils.hpp
index 0a802da..65be4e3 100644
--- a/include/sdrutils.hpp
+++ b/include/sdrutils.hpp
@@ -86,10 +86,11 @@ inline static bool getSensorSubtree(std::shared_ptr<SensorSubTree>& subtree)
                              "/xyz/openbmc_project/object_mapper",
                              "xyz.openbmc_project.ObjectMapper", "GetSubTree");
     static constexpr const auto depth = 2;
-    static constexpr std::array<const char*, 5> interfaces = {
+    static constexpr std::array<const char*, 6> interfaces = {
         "xyz.openbmc_project.Sensor.Value",
         "xyz.openbmc_project.Inventory.Item.Cpu",
         "xyz.openbmc_project.Inventory.Item.Watchdog",
+        "xyz.openbmc_project.Sensor.State",
         "xyz.openbmc_project.Sensor.Threshold.Warning",
         "xyz.openbmc_project.Sensor.Threshold.Critical"};
     mapperCall.append("/xyz/openbmc_project/sensors", depth, interfaces);
-- 
2.25.1

